python/ray/_raylet.pyx

```diff
@@ -412,14 +425,14 @@ cdef execute_task(
                     monitor_state.unregister_coroutine(coroutine)
 
                 future.add_done_callback(callback)
-
                 with nogil:
                     (core_worker.core_worker.get()
                         .YieldCurrentFiber(fiber_event))
 
-                yield future.result()
+                result = future.result()
+                return result
 
-            yield function(actor, *arguments, **kwarguments)
+            return function(actor, *arguments, **kwarguments)
```

Comment:
Why the change from yield to return here?

Reply:
Explained in comments above. `yield` mark the function generator, we don't want that.